### ISSUE TITLE
Placement and alignment of windows should not be affected by display scale

### ISSUE THREAD

**Issue by Saviq** [canonical/mir#3553]
These are two screenshots of the same surface at `--display-scale 1.5`, except one of them is moved 1 "unit" on both axes.
![image](https://github.com/user-attachments/assets/2ff71a1c-8ada-407a-80c8-345c94b4fffc)
![image](https://github.com/user-attachments/assets/d8d9a08f-d786-42fc-868b-d986cd6c6c6f)

You should easily see the second one is blurred.

The problem is that it's aligned on half-pixel (or rather, on "logical" pixel that's a square of 1.5 pixels).

You can also see this with `--display-scale 10` and trying to move a window - you'll only be able to move it in 10px increments.

**Comment by Saviq** [comment]
> some scenarios need that ("cloning" with the second display with twice the dimensions and running at scale 2).

_Originally posted by @AlanGriffiths in https://github.com/canonical/mir/issues/3555#issuecomment-2293170763_

One alternative Alan mentioned is that we could align on pixel boundary at the compositing level.

Still, I think we can have better logic for the logical grid than we do now. If there's no overlap between displays, it'd be appropriate to have the grid be the output's physical grid. Overlap makes things trickier, but we also don't need to optimize for that.

**Comment by AlanGriffiths** [comment]
> Still, I think we can have better logic for the logical grid than we do now. If there's no overlap between displays, it'd be appropriate to have the grid be the output's physical grid. Overlap makes things trickier, but we also don't need to optimize for that.

I don't think it is just about overlapping displays. Adjacent displays with different pixel density are also better handled by a logical grid.

**Comment by Saviq** [comment]
> Adjacent displays with different pixel density are also better handled by a logical grid.

Could you elaborate on that? I get it for surfaces spanning displays, but that's getting into corner cases I don't think we should optimize for.

Anyway, we need to spend proper time spec'ing this out.
